### DISCUSSION TITLE
How do I use PINIA in I18N instantiation?

### DISCUSSION THREAD

**Discussion by 1JayPeng** [vuejs/pinia#1219] [Help and Questions]
i use `vue3.2.6` `vue-i18n9.1.7` `pinia2.0.13`  
and i need use store in creatI18n()like
```
import { createI18n } from 'vue-i18n';
import { store } from '/@/store/index';
export const i18n= createI18n({
	//now is vuex
	locale: store.state.themeConfig.themeConfig.globalI18n,
	fallbackLocale: zhcnLocale.name,
	messages,
});
```
i want,and i read the [doc](https://pinia.vuejs.org/core-concepts/outside-component-usage.html#single-page-applications),but is also throw error
```
import { createI18n } from 'vue-i18n';
import {useThemeConfigStateStore} from '/@/stores/themeConfig'
export const i18n= createI18n({

	//is error ,pina not install,bur i must install
	locale: useThemeConfigStateStore().themeConfig.globalI18n,
	fallbackLocale: zhcnLocale.name,
	messages,
});
```

**Comment by dreambo8563** [comment]
@china12315   I think the error is the order issue:
it works here https://stackblitz.com/edit/github-lhblpb?file=src/main.js.

and vue-i18n provide the composition API, actually I don't think integrated it into pinia is better than `useI18n()` directly.  https://vue-i18n.intlify.dev/guide/advanced/composition.html#basic-usage

**Reply by 1JayPeng** [reply]
Thank you very much for your answer, but for the first URL you provide, because my Pinia and I8n are under different files, I still have the original issue, I agree with your point of view, essentially It does not use the configuration of the componentization API management I18n, as for the second method, according to the component-based configuration I18n, the original problem

**Reply by 1JayPeng** [reply]
In fact, I think this may be Pinia's question rather than I18N's question. This is why I ask questions under Pinia, because my I18n needs to mount it in the global, and Pinia is always not installed.

**Reply by dreambo8563** [reply]
I move the i18n and store in separate file, I am not clear if can resolve your problem
 https://stackblitz.com/edit/github-lhblpb?file=src/main.js.
